perf(language_server): use `Arc<RwLock>` instead of `Mutex` for workspace workers

Sysix opened the pull request:
`Mutex` will block others requests. `RwLock` + `Arc` is faster.
I asked ChatGPT about it:

🔒 So the Real Options Are:
✅ Arc<Mutex<Vec<WorkspaceWorker>>>

Use this if:

    All accesses (read and write) are relatively balanced or infrequent.

    You want simplicity and don't expect much read contention.

✅ Arc<RwLock<Vec<WorkspaceWorker>>>

Use this if:

    Most access is reading, and you want to allow multiple concurrent reads.

    Writes (e.g. adding/removing workers) are infrequent.

In this case, RwLock may improve performance under heavy read concurrency (e.g., lots of requests querying the current workspace list).

camc314 commented:
### Merge activity

* **May 27, 7:17 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **May 27, 7:17 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
perf(language_server): use `Arc<RwLock>` instead of `Mutex` for workspace workers (#11328)

`Mutex` will block others requests. `RwLock` + `Arc` is faster.
I asked ChatGPT about it:

🔒 So the Real Options Are:
✅ Arc<Mutex<Vec<WorkspaceWorker>>>

Use this if:

    All accesses (read and write) are relatively balanced or infrequent.

    You want simplicity and don't expect much read contention.

✅ Arc<RwLock<Vec<WorkspaceWorker>>>

Use this if:

    Most access is reading, and you want to allow multiple concurrent reads.

    Writes (e.g. adding/removing workers) are infrequent.

In this case, RwLock may improve performance under heavy read concurrency (e.g., lots of requests querying the current workspace list).
